### PR TITLE
Fix undefined variables

### PR DESCRIPTION
**Pull request by AARRRon**
This change allows you to reassign a variable that has been assigned to undefined.

### CODE REVIEW

**Review by dzjin — COMMENTED** [review]
Looks good. Can you add this to unit tests?

Can you also add how it fixes the issue? Could we have modified `backend` (whatever it is) to get the same effect?

**Review by dzjin — APPROVED** [review]

**Review by d-rep — APPROVED** [review]

File: lib/es6/activities/for.js [hunk 1]
```diff
@@ -37,7 +37,7 @@ class For extends BaseLoop {
         }
     }
 
-    _doStep(callContext, lastResult) {
+    _doStep(callContext, result, lastResult) {
```
Comment (AARRRon) [inline lib/es6/activities/for.js:40]: This fixes a failing `for` test  
```
1) Loops
       For
         should work between range 10 downto 4 by step -2:

      AssertionError: 'complete' == '1086'
      + expected - actual

      -complete
      +1086
```

**Review by greenarrowdb — APPROVED** [review]
Lookin good

**Review by greenarrowdb — APPROVED** [review]
Looks good to me

**Pull request merged**